### ISSUE TITLE
Single sample behavior 

### ISSUE THREAD

**Issue by matthew-blackman** [phetsims/projectile-data-lab#91]
In the Sampling screen, what do we want to happen in Single sample mode, when the user presses the launch button while a sample is in progress? Currently the sim hides and auto-completes the current sample (showing it on the histogram), and immediately shows an empty field with a new sample being launched.

@catherinecarter can you comment on your thoughts for the desired behavior?

**Comment by catherinecarter** [comment]
Here's an idea for "ideal" behavior, although I suspect my opinion about this behavior as "ideal" may shift after hearing others' pros and cons to this comment. I'll go through the current behavior first to help build context for myself and then move to what my brain is envisioning (this is a long one!).

**Current behavior**
Currently on the first three screens (VSM), when a user pauses the sim and the presses the Launch button again, the sim auto-unpauses and launches a new projectile. If any projectiles were frozen in the air, they are allowed to complete their path and land, with the newly launched projectile following.

The sampling screen is different in that on Single Sample mode, the cannon automatically launches a set number of projectiles (2, 5, 15, or 40) in succession even though the launch button has only been pressed once. So having the exact behavior of the VSM screens doesn't make sense since the "batch" or sample of _n_ objects matters and the launch button has different functionality on the Sampling screen.

Currently on the Sampling screen, if a user presses the launch button on Single Sample mode, then pauses prior to the current sample landing, the sample that was in the air is deleted from the sky, but its mean is still recorded in the histogram. After the current sample has been deleted from the sky, a new sample comes out and is allowed to land (provided the user hasn't again pressed the launch button, in which case, the sample would be again deleted from the sky and its mean recorded, like before).

**Ideas for improved behavior**
After thinking about the behavior listed above, it seems the most unexpected piece is when the current sample is deleted from the sky but the mean still appears on the histogram. This is confusing because the sample was deleted but the mean was still captured.

Ideally, it seems like the sample should be able to finish it's path like the previous 3 screens. While the function of the launch button is different on the VSM screen compared to the Sampling screen, the opportunity to watch anything in the air complete its path would be consistent across screens. As an example, I'll consider when _n_=15 is chosen. Mapping the visual cues on screen with the user button-pushing, this is a scenario that could happen.
- the launch button is pressed once on normal speed, and some projectiles start to come out
- the user presses pause, and the first 9 projectiles are in the air (6 still to be launched)
- the user presses the launch button _p_ additional times and the sim auto-unpauses
- the mode is automatically converted to 'fast' mode and the current sample + _p-1_ additional samples are launched quickly
- the mode is then switched back to normal speed as the final sample comes out

Alternatively, the sim could switch to the continuous mode (visually only without the toggle showing so), completes all but one sample, and any samples in between the one in the air and the last sample that would be launched would be shown on fast/continuous mode so the means showing on the histogram would make sense why they were there. The final sample launches in normal speed mode to complete the sequence.

**Idea for less ideal**
Since the sample is essentially deleted from the sky, we could just delete it and not record its mean on the histogram. Just an idea, not a suggestion for great behavior.

**Comment by matthew-blackman** [comment]
Notes from 1/23 discussion with @samreid and @catherinecarter:
- We want to try and avoid the appearance of deleting a sample when the launch button is pressed
- We want to try and avoid long chains of automated events where possible
- The intended UX of the Single sample is to see a whole sample form, and then see its mean
- The intended UX of the Continuous mode is to see samples in quick succession

In light of the above points, we agreed to try the following behavior:
- When a single sample is forming, the launch button is disabled
- The Single/Continuous launch controls should always be enabled
- Switching to Continuous mode causes the launch button to re-enable (with the Continous icon)
- Pressing the Launch button in Continuous mode causes any currently forming Single sample to auto-complete to begin the Continuous sample collection
- If the user switches to Single sample mode while a single sample is forming, the launch button re-disables.

**Comment by samreid** [comment]
I implemented the recommendations in the commit. It was straightforward except for one behavior:

* Press Launch (in single mode)
* Press Continuous

I had expected to see this start continuous mode right away (and hence the airborne projectiles would land instantly and show a mean), but I see now that was incorrect. Because the user hasn't pressed "launch" in continuous mode, the continuous mode hasn't started yet. That means the single airborne projectiles remain in the air until the user presses "launch" in continuous mode. So I think the behavior is correct, it was just unexpected.

This issue is ready for design review/testing/code review.

**Comment by matthew-blackman** [comment]
Everything looks good here. Nice work!

I agree that it seems like we could consider starting continuous launch right when the user presses the radio button, and disable the launch button completely in Continuous mode. This seems like it could even work across all screens, and may be a better UX than the stop/start button. The goal of continuous launch is to see a stream of data, so being able to stop/start it with the launch button doesn't seem like a very useful feature. It may just be adding an additional button press for the user to get to the state that they want. 

@catherinecarter and @samreid please weigh in as well!

**Comment by matthew-blackman** [comment]
Discussing this with @catherinecarter and @samreid, we agreed that the behavior for Single sample is good but the launch button re-enabling is an abrupt color change that can be distracting from the mean. I will explore some color options for the disabled state of the launch button that are less distracting while still being distinct.

**Comment by matthew-blackman** [comment]
This is ready for design review @catherinecarter. I updated the launchButtonDisabledColor. Feel free to suggest updates to that color and/or weigh in on whether this is an improvement from before.

**Comment by catherinecarter** [comment]
**Button**
I think the disabled color is good - it conveys inaction but isn't so stark in color change to distract from the mean appearing on the Sampling screen. I'd love to hear what others in design meeting think.

**Behavior**
I'd also like to address @samreid's comment above: 

> I implemented the recommendations in the commit. It was straightforward except for one behavior:
Press Launch (in single mode)
Press Continuous
I had expected to see this start continuous mode right away (and hence the airborne projectiles would land instantly and show a mean), but I see now that was incorrect. Because the user hasn't pressed "launch" in continuous mode, the continuous mode hasn't started yet. That means the single airborne projectiles remain in the air until the user presses "launch" in continuous mode. So I think the behavior is correct, it was just unexpected.

I agree that the behavior is unexpected. It makes it seem like the toggle from single to continuous does nothing except re-enable the button. And if you press it before the single sample lands, we have the same issue of it looking like the current sample is deleted.

I second what Sam expected: 
> I had expected to see this start continuous mode right away (and hence the airborne projectiles would land instantly and show a mean)

Is this possible?

**Comment by matthew-blackman** [comment]
Notes from 1/26 with @catherinecarter and @samreid:

- On the Sampling screen, we don't want to hide data that gets added to the histogram. If the user switches to Continuous during a Single launch, we should make sure that the user can see the mean for the active sample.
- We will autocomplete any active Single sample when switching to Continuous mode.
- The mean doesn't need to have a delay if the user switches into Continuous, since that mode has no delay.
- We don't want to auto-start Continuous mode from the radio buttons, since launching should feel like an action across the entire sim.

**Comment by samreid** [comment]
@matthew-blackman and I addressed the recommendation in the commit. I added a call to make sure the histogram updates and it seems to be working correctly. I think this issue could be closed after a quick spot check, then we can continue with https://github.com/phetsims/projectile-data-lab/issues/76 and the other sampling screen bugs in other issues.

**Comment by catherinecarter** [comment]
I tested it for a few minutes. The behavior looks great when I'm launching a single sample and then switch to continuous mode while the sample is in mid-air. The instant landing looks great. 

I did notice that when I pressed the launch button while in continuous mode for the first time, there seems to be two means appear on the histogram before it settles into a cadence. In the video below, focus your attention on the histogram to see what I mean:

https://github.com/phetsims/projectile-data-lab/assets/109630002/06d7132d-6aad-439c-a42a-447cbc9d5800
